CMakeLists.txt

```diff
@@ -167,6 +167,10 @@ else()
 
     message(STATUS "AMDGPU_TARGETS: ${AMDGPU_TARGETS}")
 
+    string(REGEX MATCHALL "gfx94[0-9]" AMDGPU_TARGETS_WITH_F8 "${AMDGPU_TARGETS}")
+
+    message(STATUS "AMDGPU_TARGETS_WITH_F8: ${AMDGPU_TARGETS_WITH_F8}")
+
     if( CMAKE_CXX_COMPILER_ID MATCHES "Clang" )
```

Comment:
This doesn't really work, since "AMDGPU_TARGETS" contains all supported target, not the "arch of current machine".
So even in gfx90a, ${AMDGPU_TARGETS} is still "all", which means the REGEX always match.

You can see this from the failed CI log:
http://math-ci.amd.com/blue/rest/organizations/jenkins/pipelines/main/pipelines/precheckin/pipelines/hipBLASLt/branches/PR-602/runs/21/nodes/226/steps/236/log/?start=0

Scroll down about one page and you can see :

> /var/jenkins_home/workspace/main_precheckin_hipBLASLt_PR-602/DxgGSmWjr/hipblaslt/build/release /var/jenkins_home/workspace/main_precheckin_hipBLASLt_PR-602/DxgGSmWjr/hipblaslt
> **-DAMDGPU_TARGETS=all** -DCMAKE_BUILD_TYPE=Release
> -- The CXX compiler identification is Clang 17.0.0
> -- Detecting CXX compiler ABI info
> 

And then in a later block not far from this you can you can see:

> BLIS heeader directory found: /var/jenkins_home/workspace/main_precheckin_hipBLASLt_PR-602/DxgGSmWjr/hipblaslt/build/deps/blis/include/blis
> BLIS lib found: /var/jenkins_home/workspace/main_precheckin_hipBLASLt_PR-602/DxgGSmWjr/hipblaslt/build/deps/blis/lib/libblis.a
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx90a_xnack_on
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx90a_xnack_on - Success
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx90a_xnack_off
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx90a_xnack_off - Success
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx940
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx940 - Success
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx941
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx941 - Success
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx942
> -- Performing Test COMPILER_HAS_TARGET_ID_gfx942 - Success
> **-- AMDGPU_TARGETS: gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942
> -- AMDGPU_TARGETS_WITH_F8: gfx940;gfx941;gfx942**

That's why the two examples are still built on this gfx90a CI node